ocaml-lsp-server/src/document_store.ml

```diff
@@ -190,19 +190,12 @@ let get_semantic_tokens_cache : t -> Uri.t -> semantic_tokens_cache option =
   let doc = get' t uri in
   doc.semantic_tokens_cache
 
-let change_all t ~f =
-  let all =
-    Table.foldi ~init:[] t.db ~f:(fun uri doc acc -> (uri, doc) :: acc)
-  in
-  Fiber.parallel_iter all ~f:(fun (uri, doc) ->
-      let+ doc =
-        match doc.document with
-        | None -> Fiber.return doc
-        | Some document ->
-          let+ document = f document in
-          { doc with document = Some document }
-      in
-      Table.set t.db uri doc)
```

Comment:
@ulugbekna this code was evil. We're updating a value in the hash table after waiting for the termination of `f`, but the value might be out of date by then due to a concurrent document change notification.